book/05-reporting.Rmd

```diff
@@ -1 +1,164 @@
+
+Possible output formats are:
+
+  - HTML
+  
+  - PDF
+  
+  - Word
+  
+  - Powerpoint
+  
+Which output format should be used is specified in the "YAML header" of the R Markdown document.
+
+## The YAML header
+
+In the YAML (Yet Another Markup Language) header users can specify metadata which denote the final appearance of the document.
+
+Each output format has different settings. 
+Fortunately, most settings apply to all formats.
+
+The YAML header starts and ends with three dashes: `---`.
+The **output** field is mandatory.
+
+```YAML
+---
+title: "<title>"
+author: "<author>"
+date: "`r Sys.Date()`"
+output:
+  rmarkdown::html_document:
+    toc: yes
+    number_sections: yes
+    fig_caption: yes
+    css: ../custom.css
+bibliography: lib.bib
+biblio-style: apalike
+---
+```
+
+Valid options for each output format can usually be looked up in the help page of the specific output format.
+
+For the default output format `html_document` the [R Markdown - The definitive guide](https://bookdown.org/yihui/rmarkdown/html-document.html) book is a good reference.
+
+An [R Markdown cheatsheet](https://rstudio.com/wp-content/uploads/2016/03/rmarkdown-cheatsheet-2.0.pdf) also exists.
+
+## Literate programming in R
+
+Packages {{rmarkdown}} and {{knitr}} are the base of literate programming in R.
+
+RMD's documents can be compiled
+
+- by clicking the "knit" button in RStudio (the name relates to the {{knitr}} package)
+
+- via the command line by calling `rmarkdown::render()`
+
+Behind the scenes the {{rmarkdown}} package first converts the `.Rmd` file to `.md` (markdown).
+Then [pandoc](https://pandoc.org/), which is a universal markdown converter library, converts the `.md` file to the chosen output format.
+
+### R Markdown packages
+
+The following packages are built upon {{rmarkdown}} and simplify special purposes.
+
+- [bookdown](https://bookdown.org/yihui/bookdown/): Mainly used for writing books but can also be used for reports (formats `html_document2`, `git_book`, `pdf_book`, etc.).
+
+- [thesisdown](https://github.com/ismayc/thesisdown): A package for thesis writing.
+Provides ready-to-go templates for different types and simplifies advanced LaTeX usage.
+
+- [rmdformats](https://github.com/juba/rmdformats), [pinp](https://github.com/eddelbuettel/pinp) : Different templates for literate programming documents.
+
+- [xaringan](https://github.com/yihui/xaringan): For HTML presentations via [remark.js](https://remarkjs.com).
+
+- [blogdown](https://github.com/rstudio/blogdown): For creating websites. Example: [https://pjs-web.de/](https://pjs-web.de/)
+
+- [rticles](https://github.com/rstudio/rticles): For scientific paper writing in R.
+
+### Code chunks in R Markdown
+
+To insert code into an R Markdown document, one needs to add a so called "code chunk".
+
+````
+```{<language>}
+
+```
+````
+
+This tells the document that everything within the three backticks should be interpreted as code using the given language.
```

Comment:
Would this be easier to grasp if we used a "white lie" that the chunk header is of the form ````` ```{r} ````` ?

Reply:
Umm - I'd say it's fine as it is.
Such things can fire back at some point - maybe not at us but somebody else.